### ISSUE TITLE
Map Matching (/trace_route) is returning part of the route in alternates

### ISSUE THREAD

**Issue by schettgen** [valhalla/valhalla#4802]
We doing a request to map matching api of valhalla. 
The unmatched route looks like this 
![image](https://github.com/user-attachments/assets/ab689b5b-9130-47a1-959c-3632dcbfeede)
On the top right is the start and on the bottom the stop

When we do the following request -> request.json](https://github.com/user-attachments/files/16248455/request.json)
we get the following response -> response.json](https://github.com/user-attachments/files/16248457/response.json)

## What we expect 
We would expect that valhalla is returning **the complete** map matched polyline at
`trip.legs[0].shape` and not split it into different polylines in different collections in the result.

## TL;DR What happens 
Valhalla is splitting the polyline into two. One polyline can be found in the response under 
`trip.legs[0].shape` the other is available at `alternates[0].trip.legs[0].shape`
Also the end / start of the polyline is not the same position (see below)

### Polyline under `trip.legs[0].shape`

![image](https://github.com/user-attachments/assets/94b67ad2-c28f-4932-9fec-758d4ce420d3)

```
_abv}Aek|sKjo@`dCpLl]sCbD}f@zj@aG`EfWvWxShRjQpMhf@|O|Cz@`KnDXP`@VlAl@xAlA|AzBlPjXlVzOrRjD`LlHdq@jv@fFtF`HxFbWdQdFfE`HbJjpAtzAjN`Vbt@lvA`FfLloAhmErRvy@pPnp@hFjVph@rwB|VpsBxGv`@|Pfr@bBjI|AzHxAjQZnWu@nQcDna@{@rVpB|z@dA`b@vFnjBLza@i@nZcI|rAyJn}AuF~e@uG`[wNlg@qSfd@eU~Z{t@~x@oLhPcMdRqH~OoGtSgKl`@mBfH{A|E}AjDaFjK{Qdc@gBlFoC~JoBtIuBxJuBlLgAtGyGlb@yF~]aDnQgFpR_D~KgDbKyBpGyEpN{C`I_BnDgBfCiBrAw@j@y@b@qAh@qCz@yK|B_R|DiGlBsDbBkExByE~CuChCaBrBuA|CgA~BqAlEkClJsHvYkIr_@yBzI}AzG}@~C]~@a@p@e@d@_@Va@Re@Pw@R_AJoCRwDCkESoBTcBd@}B|@aCtAiAj@gJxD_GfCsE~AeDr@yANkJf@iEJ{DMmHcA}I{AoB[cAA{@Lq@Vo@^u@r@q@v@k@z@eAxCw@pGMbBo@hGS~Bc@xCe@bBg@rAaAbAmAfAgChBs@\\y@Pu@?eG[_OwAaNk@aFu@yMeDaSuFoMgBwIm@mEKuAD{@Zw@b@_@h@Yp@WjAOnA[dF{Cde@}@lNgAfPa@lG]tGIlE@zCNpCPvBXpARx@Ph@r@vAh@x@l@v@x@t@dAv@lAf@bCp@hEr@tDh@fKhAlRxAlNv@nJVrDJtMPhNEpHIfII`J[~GYlKs@hMkAtUkCzMqBpPgDzEeArNcDtb@cMja@wMpfAk_@jyAih@fo@}T~R{G~^cLb@zDh@hDvG~n@xJj_ARpCDzBBbEAhDIjDKrFApC@xBHlBb@~DlArJ`CjPfBrLdApGtAhGnAnE|BhGpBtDhCrCpCbCrBdAbBr@gBpKoApDgDlEeKnDwMFsKFoRIwhDiAezDqA}[MuHE{\\q@ig@gC}pAyJeoA_NyhAeUuu@uKwa@iIuPmAeHIoz@QuN^oLrBwMxJcGhHoFhKmDpMyChO{AfMw@pOOtPCrO|@jUdAfPnCf[|Eta@nKtw@lFf[hLrk@dYrvAnOdt@pGb]jDzQlC`OdDjQZzDGtDY`DUjAtPpBrWlXhb@z^dQxLpQdM~U|OfMvI|J~LrAxAnArA`KvKr\\vd@lq@`eAzf@fu@vNlTvNlTfRpZ`DrGr]ls@`Y`m@x{@|gC`_@dgA~d@~qA~BzG~t@|wBbJ|XtDjM~C|PlAvKvDb_@`Jv`Alk@zaGxCx]h@tQJpLGzj@CtG@tGVfNh@bFj@nEHr@v@|GpD`TzGxa@dFh[bEzXxAnQn@pG`ArGt@|CfAtBvA|AbCbBfA`@`Ad@t@p@b@r@Vr@Hd@F|@A`Cg@lKeAxRg@pQSdJEhAWtP_@bLkA`P_Flc@sCzToBfMWlAkA`HaClKiB~E{ClGsIjOsQl[sD|G{_@py@yGrPmClIyBjIsAvG_A~Fq@zHo@jK]pFmBfZqAlSw@jQY~K[nKa@vM_@zMk@rOmBvYQ~B]rFc@|Hg@dMg@rOIfKRj_AHnw@@xXPnR\\nOn@pOp@~LfBpS`BjMlHtd@tJdn@vB~MdBlItCtJjDfHvC`F`EdFzH|GzHvF`NrJvFpEfGzFzGfIdFvHdLjSdMjQtN|NrCbCtGvFdOfMbGdGjApE^xB\\hDXjENbEPpG_n@|FePtCaShCwx@lGebBfMggAvHsEV{If@}@zEm@~BsDvHcU~b@gi@tfAaOvTiOzPoDdDeErCmEpCeEtBcE`B_FnByEjAyGxAaHv@qJn@
```

### Polyline under `alternates[0].trip.legs[0].shape`
![image](https://github.com/user-attachments/assets/1ef462b2-f3df-4f24-ab84-46318e268916)

```
k~}u}AyukmKth@ueAbU_c@rDwHl@_C|@{EzIg@rEWfgAwHdbBgMvx@mG`SiCdPuC~m@}FzCWxNaAfCQnO_AbOuAjT}Cr_@}H|nDw~@t_@aLv[{LtVcMvVaPvSmQvRsStU}YvRq[zNiYzM}ZjVon@`t@ylBrKeYtBsFnMu\\t[kz@|t@{oB|uCwwHfDoIrHwNzIgNvMyQlj@{l@fy@q|@rx@e{@dZiSjz@_g@ntAgx@f`@wUvVeNx`@uVnN{L`KmNpRa[tP_YxKyQhLkSd[cg@rQqYlGoKfC{DfF{HvCqEjOqU`^i_@lG}KrEyQvD_gAF}AnDuiAtBqk@zByg@n@sI|MsiA~AwMrEe_@hCwTnAqLnFia@`AsIfBuMtKk~@dIwo@xY}~BlCaP`C_M`CaLrCwL~@gDcByEy@}A_A{AsBeC}AkBcFuCsBqDqCwF{AuDuBgGkBqGe@sD[mDSwDG}BG_FEaM`[sCrHw@tDg@xEgAnD{@fDuAhEyBbEmBvFyDdD}BjBqAlBgA`B}@nB}@hBq@nBk@tA[dBSdCS`CElBB`CLhCV`Cd@z@TlBl@nD~A|BvA`BnAfB`BfCdC|CjExBdEbApBpB|ElBvFbBzF~A|GjBnI`DxPpGv\\pDzRvIxd@xIje@nGd\\lD~PvCbNhElSpIxb@fL`m@lDlRhDjT~ClS~B|NpD`QpDpPvCfLdCtHrB~FjC~FfEtIpD|GjHrLtRtZvVn`@t[xf@dSnZzCjE~CpD|CdDnCtC`FdEhGlFpE`DpDbCjE|BbEbBvDhApFfA|CPlCD`CGlCa@jCq@lHiChKeElFwAlCi@xAI`BG`BN|Af@`B|@nA`AvA`Bt@pAZv@b@rAt@~C\\pCXdCJtC?nD~HzAlHnAdDHbBIlCc@pC{@dCcBzCwCrA_B
```

## Polyline have different starting points
Detail view of polyline from  `trip.legs[0].shape`
![image](https://github.com/user-attachments/assets/2e6ec3e0-cd59-4b9a-b1e7-c77fd8952f98)

Detail view of polyline from  `alternates[0].trip.legs[0].shape`
![image](https://github.com/user-attachments/assets/d47056a4-6da4-4439-8f47-e7947be2b2f9)
Start point should be at (1)
 
 The polylines are not ending / starting at least at the same point

**Comment by johannes-no** [comment]
Hello @schettgen ,
i'm honestly a little confused as route responses and trace attribute requests are mixed here.
The response.json seems to be the response of a route request and not a trace attribute request. 
Could you please check this again and rerun the trace request with the filter "shape"?

**Comment by chrstnbwnkl** [comment]
I think request and response do match, the end point in question is not `/trace_attributes` but `/trace_route`, which returns a response in the same format as `/route`

**Comment by johannes-no** [comment]
Oh sorry my fault!
Unfortunately I can not reproduce the error but we do a different U-Turn in Tawern:
![image](https://github.com/user-attachments/assets/279374aa-d5ee-4b0d-b653-3f4a33cfb248)

**Comment by chrstnbwnkl** [comment]
I can reproduce this using the public Valhalla instance. I had a quick look and it turns out this is a feature, not a bug: in case of discontinuities, multiple paths are returned as alternatives. We should at least document this behavior somewhere.

**Comment by chrstnbwnkl** [comment]
Whoops, it's actually documented, just deep in the Meili docs.

**Comment by kevinkreiser** [comment]
yeah it didnt always work this way. for the longest time we didnt have a way to return alternates in the regular route response. once i added that this kind of just happened as a side effect. before that if there was a discontinuity on the regular route it would just throw an error and give you nothing. this, while not exactly intuitive seemed better
